### ISSUE TITLE
compilation fails with WebRTC on windows using vcpkg

### ISSUE THREAD

**Issue by anchor76** [ValveSoftware/GameNetworkingSockets#140]
it seems the current head fails during compilation, if i want it with WebRTC.
the deafult config (without WebRTC) compiles flawlessly.

log file:

[install-x64-windows-dbg-out.log](https://github.com/ValveSoftware/GameNetworkingSockets/files/5313054/install-x64-windows-dbg-out.log)

error messages:

```
..\..\..\..\src\external\abseil\absl\hash\internal\hash.cc(51): error C2491: 'absl::lts_2020_02_25::hash_internal::CityHashState::kSeed': definition of dllimport static data member not allowed 
..\..\..\..\src\external\abseil\absl\numeric\int128.cc(28): error C2491: 'absl::lts_2020_02_25::kuint128max': definition of dllimport data not allowed 

```

platform: win10 x64 (tried on win7 x64 too with the same result)
compiler: visual studio 2019 (command prompt)
command: .\vcpkg\vcpkg --overlay-ports=vcpkg_ports install gamenetworkingsockets --triplet x64-windows

edit: i tried to modify ABSL_DLL macro to make it empty, but then it fails during installation with the error:

`CMake Error at scripts/cmake/vcpkg_fixup_cmake_targets.cmake:72 (message):

  'C:/!/GameNetworkingSockets/vcpkg/packages/gamenetworkingsockets_x64-windows/debug/lib/cmake/GameNetworkingSockets'
  does not exist.`

**Comment by fletcherdvalve** [comment]
Thanks, I'll check

**Comment by anchor76** [comment]
hello @fletcherdvalve ,

excuse me for pinging this issue, but would be very good to fix this compile problem somehow.
i would appreciate any advice.

thanks ;)

**Comment by fletcherdvalve** [comment]
Well, I'm having trouble downloading and installing protobuf through vcpkg, so I cannot reproduce the problem at the moment.

Building stuff on Windows is just the absolute worst.

**Comment by anchor76** [comment]
meanwhile we managed to compile **GameNetworkingSockets** on windows by adding the required sources into a vs2019 project, and configuring it with hand. (it was a long process)

_maybe we should rename the issue to reflect the problem is with the vcpkg way._

**Comment by fletcherdvalve** [comment]
OK, I'm not reproducing this problem.  I am using VS2019, x64.  I compiled and "installed" protobuf (since I couldn't get vcpkg to download it) and then I ran cmake with this command line:

```
cmake .. -G Ninja -D CMAKE_PREFIX_PATH=c:\sdk\protobuf-amd64 -DUSE_STEAMWEBRTC=ON
```

I didn't need to manually create any project files or anything -- I don't think you should have to do that.

So I think you re right that it may be specific to using vcpkg.  Unfortunately, because of some issues with vcpkg not being able to download protobuf, I wasn't able to reproduce exactly what you got, to investigate.  vcpkg is basically just busted right now.  I'm not really an expert on vcpkg, so I'm not sure who to blame there.

Ug Windows.

**Comment by fletcherdvalve** [comment]
Looks like the root cause of the vcpkg problem I hit is this:

https://github.com/microsoft/vcpkg/issues/13286

**Comment by fletcherdvalve** [comment]
I renamed the issue.  Once the upstream vcpkg msys issues are sorted out, I'll repro the WebRTC specific one and see what I can do.

**Comment by aiusepsi** [comment]
I had a gander at this, #143 adds a [webrtc] option to the vcpkg port which uses the vcpkg port of abseil, which is in keeping with the general advice for vcpkg to use other ports rather than bundling them.

**Comment by fletcherdvalve** [comment]
Great!  Thanks for looking into this.

While you are in there, can you see if you can fix #138 as well?

**Comment by aiusepsi** [comment]
Should also fix #138, with one slight edge case that the install target won't create a config file to allow import from another CMake project if the build used the abseil submodule, rather than pulling in an abseil package from elsewhere. 

Not sure what the best way to resolve that is, but at least it won't error out now.

**Comment by fletcherdvalve** [comment]
@aiusepsi Thanks for this.  One issue is that LInux is now broken.  Let me know if you have some time to take a look.  I just merged your change in the github UI without testing anything.

**Comment by aiusepsi** [comment]
@fletcherdvalve Tried to replicate with a fresh Ubuntu docker image and it seems fine. Wouldn't entirely rule out some variance in CMake behaviour, though. What version of Ubuntu are you using? I can try again to replicate.

**Comment by fletcherdvalve** [comment]
Thanks I'll see what's up tomorrow!

**Comment by fletcherdvalve** [comment]
I think it might be a version issue.  I am using cmake 3.10.2, which is not SUUUper old, it's recent enough that I think we should support it.  Do you know enough cmake-foo to get the older cmake to work?  (Can we specify the install step in the CMakeLisrs.txt file in the subdirectory, instead of adding it to the root one?)

https://stackoverflow.com/questions/34443128/cmake-install-targets-in-subdirectories

```
CMake Error at src/CMakeLists.txt:299 (install):
  install TARGETS given target "steamwebrtc" which does not exist in this
  directory.
```

**Comment by fletcherdvalve** [comment]
Having a discussion with @tycho right now about minimum required cmake version.  We definitely should have a travis config that exercises the oldest cmake we support.  Only question is what version we want to make the minimum.

**Comment by bgilmore** [comment]
I would advocate for pinning the lowest supported CMake version at 3.10 since it's the version that's available in Ubuntu 18.04 without having to reach for backports/PPAs.

**Comment by fletcherdvalve** [comment]
This is fixed in 5d479a081996d26ef0d4d46d344de01a048fa0eb
